ccl/multiregionccl: TestIndexDescriptorUpdateForImplicitColumns failed

ccl/multiregionccl.TestIndexDescriptorUpdateForImplicitColumns [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3148628&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3148628&tab=artifacts#/) on master @ [f74a1684f51f1d9a1f4872a849ca519736646d7e](https://github.com/cockroachdb/cockroach/commits/f74a1684f51f1d9a1f4872a849ca519736646d7e):

Fatal error:

```
panic: collecting referenced descriptors: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1625480090.955816825,0 encountered previous write with future timestamp 1625480090.955816825,1 within uncertainty interval ``t <= (local=1625480090.959494638,0, global=1625480091.455816825,0)``; observed timestamps: [{1 1625480090.955816825,0} {2 1625480090.959494638,0}]: "unnamed" meta={id=921933b5 pri=0.00445675 epo=0 ts=1625480090.955816825,0 min=1625480090.955816825,0 seq=0} lock=false stat=PENDING rts=1625480090.955816825,0 wto=false gul=1625480091.455816825,0 [recovered]
	panic: collecting referenced descriptors: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1625480090.955816825,0 encountered previous write with future timestamp 1625480090.955816825,1 within uncertainty interval ``t <= (local=1625480090.959494638,0, global=1625480091.455816825,0)``; observed timestamps: [{1 1625480090.955816825,0} {2 1625480090.959494638,0}]: "unnamed" meta={id=921933b5 pri=0.00445675 epo=0 ts=1625480090.955816825,0 min=1625480090.955816825,0 seq=0} lock=false stat=PENDING rts=1625480090.955816825,0 wto=false gul=1625480091.455816825,0

```
Stack: 

```
goroutine 429224 [running]:
testing.tRunner.func1.2(0x6db2200, 0xc009143f80)
	/usr/local/go/src/testing/testing.go:1143 +0x49f
testing.tRunner.func1(0xc004109800)
	/usr/local/go/src/testing/testing.go:1146 +0x695
panic(0x6db2200, 0xc009143f80)
	/usr/local/go/src/runtime/panic.go:971 +0x499
github.com/cockroachdb/cockroach/pkg/sql/catalog/catalogkv.testingGetObjectDescriptor.func1(0xaa4f780, 0xc0001a6010, 0xc003c2a420, 0xc000eae500, 0xc000eae500)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/catalogkv/test_utils.go:155 +0x505
github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn.func1(0xaa4f780, 0xc0001a6010, 0xc003c2a420, 0xc000eae500, 0x2855200)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:772 +0x5f
github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec(0xc003c2a420, 0xaa4f780, 0xc0001a6010, 0xc00a4a5bf8, 0xc003c2a420, 0xaa0a580)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:870 +0xef
github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn(0xc003648b80, 0xaa4f780, 0xc0001a6010, 0xc0030b7c58, 0x0, 0xaa0a580)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:771 +0x125
github.com/cockroachdb/cockroach/pkg/sql/catalog/catalogkv.testingGetObjectDescriptor(0xc003648b80, 0xc0001c1d58, 0xc0001c1d58, 0x0, 0x724c5ad, 0x4, 0x72523e3, 0x6, 0x7249b4a, 0x2, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/catalogkv/test_utils.go:133 +0x157
github.com/cockroachdb/cockroach/pkg/sql/catalog/catalogkv.TestingGetImmutableTableDescriptor(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/catalogkv/test_utils.go:51
github.com/cockroachdb/cockroach/pkg/sql/catalog/catalogkv.TestingGetTableDescriptor(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/catalogkv/test_utils.go:43
github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test.TestIndexDescriptorUpdateForImplicitColumns.func1(0x7249b4a, 0x2, 0xc004109800, 0x73496c5, 0x35)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/regional_by_row_test.go:913 +0x145
github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test.TestIndexDescriptorUpdateForImplicitColumns.func3(0xc004109800)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/regional_by_row_test.go:962 +0x869
testing.tRunner(0xc004109800, 0xc00df2c978)
	/usr/local/go/src/testing/testing.go:1193 +0x203
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1238 +0x5d8
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestIndexDescriptorUpdateForImplicitColumns
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestIndexDescriptorUpdateForImplicitColumns748556575
    test_log_scope.go:74: use -show-logs to present logs inline
--- FAIL: TestIndexDescriptorUpdateForImplicitColumns (40.62s)
=== RUN   TestIndexDescriptorUpdateForImplicitColumns/secondary_index
    --- FAIL: TestIndexDescriptorUpdateForImplicitColumns/secondary_index (10.98s)
```
</p>
</details>
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestIndexDescriptorUpdateForImplicitColumns PKG=./pkg/ccl/multiregionccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
/cc @cockroachdb/sql-queries postamar
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestIndexDescriptorUpdateForImplicitColumns.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/catalog/catalogkv/test_utils.go
+++ b/pkg/sql/catalog/catalogkv/test_utils.go
@@ -19,6 +19,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/tabledesc"
+	"github.com/cockroachdb/errors"
 )
 
 // TestingGetTableDescriptorFromSchema retrieves a table descriptor directly
@@ -133,28 +134,28 @@ func testingGetObjectDescriptor(
 	if err := kvDB.Txn(ctx, func(ctx context.Context, txn *kv.Txn) (err error) {
 		found, dbID, err := LookupDatabaseID(ctx, txn, codec, database)
 		if err != nil {
-			panic(err)
-		} else if !found {
-			panic(fmt.Sprintf("database %s not found", database))
+			return err
+		}
+		if !found {
+			return errors.Errorf("database %s not found", database)
 		}
 		exists, schemaID, err := ResolveSchemaID(ctx, txn, codec, dbID, schema)
 		if err != nil {
-			panic(err)
-		} else if !exists {
-			panic(fmt.Sprintf("schema %s not found", schema))
+			return err
+		}
+		if !exists {
+			return errors.Errorf("schema %s not found", schema)
 		}
 		found, objectID, err := LookupObjectID(ctx, txn, codec, dbID, schemaID, object)
 		if err != nil {
-			panic(err)
-		} else if !found {
-			panic(fmt.Sprintf("object %s not found", object))
+			return err
+		}
+		if !found {
+			return errors.Errorf("object %s not found", object)
 		}
 		desc, err = getDescriptorByID(
 			ctx, txn, codec, objectID, immutable, catalog.Any, true /* required */)
-		if err != nil {
-			panic(err)
-		}
-		return nil
+		return err
 	}); err != nil {
 		panic(err)
 	}

